### DISCUSSION TITLE
Problems with .NET Native compilation

### DISCUSSION THREAD

**Discussion by cmpalmer66** [dlemstra/Magick.NET#1578] [Help]
We have a Xamarin.Forms application targeting Android, iOS, and UWP. For the UWP build only, we have a need for Magick.NET. Unfortunately, to put the app in the Windows Store, it must be compiled with the .NET Native compiler.

The working app without Magick.NET compiles and runs fine and is currently in the Store.

But when we add the Magick.NET Nuget package, the app runs and works fine locally (without .NET native), but when compiled natively, it's broken in several weird ways that don't involve calling Magick.NET. Basically, everything that uses Reflection or P/Invoke is broken.

If we remove the Magic.NET reference, it works again.

We also tried referencing the DLLs directly in our project and, again, it works great running locally without .NET native. But when we compile with the .NET Native compiler, the app won't even open (also again, long before it gets to the Magick.NET calls).

I've tried doing some things with the rd.xml file, but it doesn't seem to have any effect.

I realize we're an edge case, but has anyone seen similar problems or done this without errors? Basically UWP with Magic.NET compiled with the .NET Native compiler (we're using Azure Pipelines and targeting x64),

**Comment by dlemstra** [comment]
I don't think it is possible to perform .NET Native compilation with this library.
